autogpt_platform/backend/backend/blocks/xquik/search_test.py

```
@@ -0,0 +1,212 @@
+        query="AutoGPT",
+    )
+
+    with patch.object(
+        block,
+        "_search",
+        new=AsyncMock(side_effect=RuntimeError("upstream unavailable")),
+    ):
+        with pytest.raises(
+            BlockExecutionError,
+            match="Xquik search failed: upstream unavailable",
+        ):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)